src/components/Collection/SeriesMetadata.tsx

```diff
@@ -3,58 +3,47 @@ import { mdiCloseCircleOutline, mdiOpenInNew, mdiPencilCircleOutline, mdiPlusCir
 import { Icon } from '@mdi/react';
 
 import Button from '@/components/Input/Button';
-import { useDeleteSeriesTvdbLinkMutation } from '@/core/react-query/series/mutations';
+import { useDeleteSeriesTmdbLinkMutation, useDeleteSeriesTvdbLinkMutation } from '@/core/react-query/series/mutations';
 import useEventCallback from '@/hooks/useEventCallback';
 
-const MetadataLink = (
-  { id, seriesId, site, type }: {
-    id?: number | number[] | null;
-    seriesId: number;
-    site: 'AniDB' | 'TMDB' | 'TvDB' | 'TraktTv';
-    type?: 'Movie' | 'Show';
-  },
-) => {
-  const linkId = Array.isArray(id) ? id[0] || null : id ?? null;
+type Props = {
+  id?: number;
+  seriesId: number;
+  site: 'AniDB' | 'TMDB' | 'TvDB' | 'TraktTv';
+  type?: 'Movie' | 'Show';
+};
 
+const MetadataLink = ({ id, seriesId, site, type }: Props) => {
+  const { mutate: deleteTmdbLink } = useDeleteSeriesTmdbLinkMutation(type ?? 'Movie');
   const { mutate: deleteTvdbLink } = useDeleteSeriesTvdbLinkMutation();
 
   const siteLink = useMemo(() => {
-    if (!linkId) return '#';
+    if (!id) return '#';
     switch (site) {
       case 'AniDB':
-        return `https://anidb.net/anime/${linkId}`;
+        return `https://anidb.net/anime/${id}`;
       case 'TMDB':
-        switch (type) {
-          case 'Show':
-            return `https://www.themoviedb.org/tv/${linkId}`;
-          default:
-            return `https://www.themoviedb.org/movie/${linkId}`;
-        }
+        return `https://www.themoviedb.org/${type === 'Show' ? 'tv' : 'movie'}/${id}`;
       case 'TvDB':
-        return `https://thetvdb.com/?tab=series&id=${linkId}`;
+        return `https://thetvdb.com/?tab=series&id=${id}`;
       case 'TraktTv':
         // TODO: Figure how to get trakt series link using ID
         return '#';
       default:
         return '#';
     }
-  }, [linkId, site, type]);
+  }, [id, site, type]);
 
-  const canDisable = site === 'TvDB' || site === 'TMDB';
+  const canRemoveLink = useMemo(() => site === 'TvDB' || site === 'TMDB', [site]);
 
-  const disableMetadata = useEventCallback(() => {
-    if (!linkId) return;
+  const removeLink = useEventCallback(() => {
+    if (!id) return;
     switch (site) {
       case 'TvDB':
         deleteTvdbLink(seriesId);
         break;
       case 'TMDB':
-        switch (type) {
-          case 'Show':
-            break;
-          default:
-            break;
-        }
+        deleteTmdbLink(seriesId);
```

Comment:
i think you just broke it for movies. \*_looks a but further down_\* nope, but please rename the helper so it's less confusing that it works for both shows and movies. 😔

Reply:
It would be confusing if it mentioned one or the other.
The current one just means it deletes *a* tmdb link, doesn't matter which one